tower/src/reconnect/mod.rs

```diff
@@ -146,6 +146,15 @@ where
         let fut = service.call(request);
         ResponseFuture::new(fut)
     }
+
+    fn disarm(&mut self) {
+        // we must be in State::Connected if poll_ready succeeded
+        if let State::Connected(ref mut service) = self.state {
+            service.disarm()
```

Comment:
technically should be able to disarm the make service?

Reply:
I don't think so? `disarm` should only ever be called after `poll_ready` completes successfully, and if I'm reading the code right, that always implies that the state is `State::Connected`. Which in turn implies that we called `poll_ready` on the make service _and_ called `make_service` (which consumes that readiness).

Reply:
that is fair!